src/commands/logs.rs

```diff
@@ -0,0 +1,120 @@
+use anyhow::{anyhow, Result};
+
+use crate::commands::deploy::{login_connection, SPIN_DEPLOY_CHANNEL_NAME};
+use crate::commands::login::LoginConnection;
+use crate::commands::{get_app_id_cloud, get_channel_id_cloud};
+use crate::opts::*;
+use clap::Parser;
+use cloud::client::{Client as CloudClient, ConnectionConfig};
+use uuid::Uuid;
+
+/// Fetch and tail logs of an application deployed to the Fermyon Cloud.
+#[derive(Parser, Debug)]
+#[clap(about = "fetch logs for an app from Fermyon Cloud")]
+pub struct LogsCommand {
+    /// Find app to fetch logs for in the Fermyon instance saved under the specified name.
+    /// If omitted, Spin looks for app in default unnamed instance.
+    #[clap(
+        name = "environment-name",
+        long = "environment-name",
+        env = DEPLOYMENT_ENV_NAME_ENV
+    )]
+    pub deployment_env_id: Option<String>,
+
+    /// App name
+    #[clap(name = "app-name", long = "app-name")]
+    pub app_name: String,
+
+    /// Follow logs output
+    #[clap(name = "follow", long = "follow")]
+    pub follow: bool,
+
+    /// Number of lines to show from the end of the logs
+    #[clap(name = "tail", long = "tail", default_value = "10")]
+    pub lastn: usize,
+
+    /// interval in secs to refresh logs from cloud
+    #[clap(name = "interval", long = "interval", default_value = "5")]
+    pub interval_secs: u64,
+}
+
+impl LogsCommand {
+    pub async fn run(self) -> Result<()> {
+        let login_connection = login_connection(self.deployment_env_id.as_deref()).await?;
+        let app_name: String = self.app_name.clone();
+        self.tail_logs(login_connection, app_name)
+            .await
+            .map_err(|e| anyhow!("{:?}\n\nLearn more at {}", e, "DEVELOPER_CLOUD_FAQ"))
+    }
+
+    async fn tail_logs(self, login_connection: LoginConnection, app_name: String) -> Result<()> {
+        let connection_config = ConnectionConfig {
+            url: login_connection.url.to_string(),
+            insecure: login_connection.danger_accept_invalid_certs,
+            token: login_connection.token.clone(),
+        };
+
+        let client = CloudClient::new(connection_config.clone());
+        let app_id = get_app_id_cloud(&client, app_name.as_str()).await?;
```

Comment:
If you create the client in `run` using `create_cloud_client`, you can pass it and simplify this
```suggestion
    async fn tail_logs(self, client: &CloudClient, app_name: String) -> Result<()> {
        let app_id = get_app_id_cloud(client, app_name.as_str()).await?;
```

Reply:
👍  great catch. thank you.